name-registry/src/main.sw

```diff
@@ -0,0 +1,165 @@
+contract;
+
+dep data_structures;
+dep errors;
+dep events;
+dep interface;
+
+use data_structures::Record;
+use errors::Errors;
+use events::{IdentityChanged, NameRegistered, OwnerChanged, RegistrationExtended};
+use interface::NameRegistry;
+use std::{
+    block::timestamp,
+    chain::auth::msg_sender,
+    constants::{
+        BASE_ASSET_ID,
+    },
+    context::call_frames::msg_asset_id,
+    context::msg_amount,
+    logging::log,
+    option::Option,
+    result::Result,
+    revert::{
+        require,
+        revert,
+    },
+    storage::StorageMap,
+};
+
+storage {
+    /// A mapping of names to an option of records, with a none representing an unregistered name
+    names: StorageMap<str[8], Option<Record>> = StorageMap {},
+}
+
+/// The amount to charge per hundred seconds per name
+const PRICE_PER_HUNDRED: u64 = 1;
+
+impl NameRegistry for Contract {
+    #[storage(read)]
+    fn expiry(name: str[8]) -> u64 {
+        match storage.names.get(name) {
+            Option::Some(record) => {
+                record.expiry
+            },
+            None => {
+                log(Errors::NameNotRegistered);
+                revert(0)
+            }
+        }
+    }
+
+    #[storage(read, write)]
+    fn extend(name: str[8], duration: u64) {
```

Comment:
Seems like anyone can extend the duration. Do we want this to be just the owner?

Reply:
I dont see why, a person may want to use a different address to pay the fees, and even if a complete stranger pays the fees it does not harm the owner in any way